Gems/ROS2/Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -112,6 +113,18 @@ namespace ROS2
         return requests;
     }
 
+    uint8_t LidarRaycaster::GetClassIdForEntity(AZ::EntityId entityId)
+    {
+        if (auto it = m_entityIdToClassIdCache.find(entityId); it != m_entityIdToClassIdCache.end())
```

Comment:
I will run it again but from what I remember this code performed very poorly without the caching.

Reply:
I've run a simple benchmark and it seems that the cache can be removed.

Reply:
I've run a benchmark of both solutions with 1 lidar, 256+ entities (with ~200 lidar raycasts performed) and those are the results:
![image](https://github.com/user-attachments/assets/c051b870-d4f8-4d63-823c-4021599d21b7)
![image](https://github.com/user-attachments/assets/c3f97a35-af4b-459f-bd0e-f8918bb679fa)
The difference is slim but I think we should keep it since the Scene Queries raycaster is already slow.

Reply:
Let keep it. Thanks for test!